rosbag2_transport/src/rosbag2_transport/recorder.hpp

```diff
@@ -101,6 +114,7 @@ class Recorder
   std::unordered_map<std::string, std::shared_ptr<GenericSubscription>> subscriptions_;
   std::unordered_set<std::string> topics_warned_about_incompatibility_;
   std::string serialization_format_;
+  YAML::Node qos_profile_overrides_;
```

Comment:
This should be a map at this point - we should be storing structured data, the C++ YAML node is an intermediate format